walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,99 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    /* request is longer than a whole day, don't return any slots */
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
```

Comment:
Maybe add a check to ensure that `request.getDuration()` is positive. Or is that enforced elsewhere?

Reply:
Yes, the html form where user specifies duration does not allow values less than zero

Reply:
I still think you should add a check here. The general rule is that you can never trust frontend code. Since it runs on the user side, they can potentially modify it and cause the servlet/backend to crash.
All user inputs/requests should be treated as untrustworthy.

E.g. I can just remove the min="0" tag using the dev tools and issue an invalid request:

![image](https://user-images.githubusercontent.com/67581577/89391640-c307cd00-d708-11ea-8fc9-aaff62c141d8.png)